datacube/index/_spatial.py

```diff
@@ -29,7 +29,7 @@ def strip_all_spatial_fields_from_query(q: QueryDict) -> QueryDict:
     }
 
 
-def extract_geom_from_query(**q: QueryField | tuple) -> Geometry | None:
+def extract_geom_from_query(**q: QueryDict | tuple | None) -> Geometry | None:
```

Comment:
Will this function not crash on line 42 if one passes in a tuple?

The same thing will happen if passing in None, but I'm trying to fix a couple of type annotations with this PR, old functional bugs should be fixed in a separate PR.

Reply:
Now that Paul's explained `**kwargs` expansions to us, the correct args/type should be `**q: QueryField`. It's not possible for it to be a tuple, so it's also not possible for it to crash.